From dddc5f5a79e8d80c1888b6de777bec1c98b57216 Mon Sep 17 00:00:00 2001
From: Martin Kutlak <mkutlak@redhat.com>
Date: Fri, 12 Jan 2018 12:58:10 +0100
Subject: [PATCH] reporter-mailx: rely on configured email

As a side effect of the fix of rhbz#1093375, the users loosed the control
over their reporter-mailx via configuration file:
/etc/libreport/plugins/mailx.conf

This fix retains this option for the user, but still solves
rhbz#1093375.

Related to rhbz#1323625

Signed-off-by: Martin Kutlak <mkutlak@redhat.com>
---
 src/plugins/mailx.conf       | 8 ++++----
 src/plugins/mailx_event.conf | 8 --------
 src/plugins/reporter-mailx.c | 2 +-
 3 files changed, 5 insertions(+), 13 deletions(-)

diff --git a/src/plugins/mailx.conf b/src/plugins/mailx.conf
index 3b23583..7c8fcae 100644
--- a/src/plugins/mailx.conf
+++ b/src/plugins/mailx.conf
@@ -4,9 +4,9 @@
 # and you don't want to specify parameters in every tool invocation.
 #
 # String parameters:
-# Subject=
-# EmailFrom=
-# EmailTo=
-#
+Subject="[abrt] a crash has been detected"
+EmailFrom="ABRT Daemon <DoNotReply>"
+EmailTo="root@localhost"
+
 # Boolean parameter:
 # SendBinaryData=yes/no
diff --git a/src/plugins/mailx_event.conf b/src/plugins/mailx_event.conf
index 88c53ec..d3b6db0 100644
--- a/src/plugins/mailx_event.conf
+++ b/src/plugins/mailx_event.conf
@@ -1,15 +1,7 @@
 EVENT=notify
-    # do not rely on the default config nor on the config file
-    Mailx_Subject="[abrt] a crash has been detected" \
-    Mailx_EmailFrom="ABRT Daemon <DoNotReply>" \
-    Mailx_EmailTo="root@localhost" \
     reporter-mailx --notify-only
 
 EVENT=notify-dup
-    # do not rely on the default config nor on the config file
-    Mailx_Subject="[abrt] a crash has been detected again" \
-    Mailx_EmailFrom="ABRT Daemon <DoNotReply>" \
-    Mailx_EmailTo="root@localhost" \
     reporter-mailx --notify-only
 
 # Users should user rhtsupport plugin, so disabling others by default
diff --git a/src/plugins/reporter-mailx.c b/src/plugins/reporter-mailx.c
index 710fe2b..c19150c 100644
--- a/src/plugins/reporter-mailx.c
+++ b/src/plugins/reporter-mailx.c
@@ -66,7 +66,7 @@ static void create_and_send_email(
     env = getenv("Mailx_Subject");
     const char *subject = (env ? env : get_map_string_item_or_NULL(settings, "Subject") ? : "[abrt] full crash report");
     env = getenv("Mailx_EmailFrom");
-    const char *email_from = (env ? env : get_map_string_item_or_NULL(settings, "EmailFrom") ? : "user@localhost");
+    const char *email_from = (env ? env : get_map_string_item_or_NULL(settings, "EmailFrom") ? : "ABRT Daemon <DoNotReply>");
     env = getenv("Mailx_EmailTo");
     const char *email_to = (env ? env : get_map_string_item_or_NULL(settings, "EmailTo") ? : "root@localhost");
     env = getenv("Mailx_SendBinaryData");
-- 
2.14.3

